File: admin/server/router/ddz/game_config.go
package ddz

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DDZGameConfigRouter struct{}

func (r *DDZGameConfigRouter) InitDDZGameConfigRouter(Router *gin.RouterGroup) {
	ddzGameConfigRouter := Router.Group("ddz/gameConfig").Use(middleware.OperationRecord())
	ddzGameConfigRouterWithoutRecord := Router.Group("ddz/gameConfig")

	ddzGameConfigApi := v1.ApiGroupApp.DDZApiGroup.DDZGameConfigApi
	{
		ddzGameConfigRouter.POST("create", ddzGameConfigApi.CreateGameConfig)   // 创建游戏配置
		ddzGameConfigRouter.PUT("update", ddzGameConfigApi.UpdateGameConfig)    // 更新游戏配置
		ddzGameConfigRouter.DELETE("delete", ddzGameConfigApi.DeleteGameConfig) // 删除游戏配置
	}
	{
		ddzGameConfigRouterWithoutRecord.POST("list", ddzGameConfigApi.GetGameConfigList) // 获取游戏配置列表
	}
}

File: admin/web/src/api/ddz/gameConfig.js
import adminService from '@/utils/adminRequest'

export const getGameConfigList = (data) => {
  return adminService({
    url: '/ddz/gameConfig/list',
    method: 'post',
    data
  })
}

export const createGameConfig = (data) => {
  return adminService({
    url: '/ddz/gameConfig/create',
    method: 'post',
    data
  })
}

export const updateGameConfig = (data) => {
  return adminService({
    url: '/ddz/gameConfig/update',
    method: 'put',
    data
  })
}

export const deleteGameConfig = (data) => {
  return adminService({
    url: '/ddz/gameConfig/delete',
    method: 'delete',
    data
  })
}
